pages/api/zapbye.ts

```diff
@@ -0,0 +1,100 @@
+                  if (err) {
+                        salesforceConnection = null;
+                        res.statusCode = 500;
+                        res.body = err;
+                        return console.error(err);
+                  }
+            });
+      }
+      let lemlistPayload = req.body;
+      await salesforceConnection.sobject(CONSTANTS.SALESFORCE_SOBJECT_TYPE).create({
+            TaskSubtype: CONSTANTS.SALESFORCE_TASK_SUBTYPE,
+            Subject: CONSTANTS.SALESFORCE_TASK_SUBJECT,
+            Status: CONSTANTS.SALESFORCE_STATUS,
+            RecordTypeId: CONSTANTS.SALESFORCE_LEMLISTTASK_ID,
+            ActivityDate: new Date(), //DueDate
+            Lemlist_Campaign_Name__c: await getLemlistCampaignNameFromCampaignId(lemlistPayload.campaignId),
+            Lemlist_Sender__c: lemlistPayload.leadFirstName + ' ' + lemlistPayload.leadLastName,
+            Lemlist_Email__c: lemlistPayload.leadEmail,
+            Lemlist_Task_Type__c: lemlistPayload.type,
+            Lemlist_Sequence_Step_Number__c: lemlistPayload.sequenceStep,
+      },{},function(err, ret) {
+            if (err || !ret.success) {
+                  return console.error(err, ret);
+            }
+            console.log('Task created successfully: ', ret);
+      });
+      res.statusCode = 200;
+      res.json({ response: "200 OK" });
+}
+
+interface LemlistCampaign {
+      _id: string,
+      name: string,
+      archived: boolean
+}
+
+async function getLemlistCampaigns(): Promise<JSON> {
+      let endpoint = "https://:" + CONSTANTS.LEMLIST_API_KEY + "@api.lemlist.com/api/campaigns";
+      console.log("Attempting to fetch Lemlist Campaigns from API");
+      const response: Response = await fetch(endpoint);
+      return response.json();
+}
+
+async function getLemlistCampaignNameFromCampaignId(campaignId: string): Promise<string> {
+      let campaign = filterCampaignById(cachedCampaignsArray, campaignId);
+      if(!campaign) {
+            //we keep track of how long ago we last pulled from Lemlist so we don't end up getting rate limited if there's an ID we can't find
+            let now = Date.now();
+            if(!lastCampaignCheckedTimestamp || now - lastCampaignCheckedTimestamp > CONSTANTS.LEMLIST_CAMPAIGN_FETCH_COOLDOWN_TIME_IN_MS) {
+                  cachedCampaignsArray = await getLemlistCampaigns() as unknown as Array<LemlistCampaign>;
+                  campaign = filterCampaignById(cachedCampaignsArray, campaignId);
+                  lastCampaignCheckedTimestamp = now;
+            }
+      }
+      return campaign;
+}
+
+function filterCampaignById(campaigns: Array<LemlistCampaign>, campaignId: string): string {
+      for (let i = 0; i < campaigns.length; i++) {
+            if (campaigns[i]._id === campaignId) {
+                  return campaigns[i].name;
+            }
+      }
+      return null;
+}
```

Comment:
```suggestion
const filterCampaignById = (campaigns: Array<LemlistCampaign>, campaignId: string): string => {
      for (const i in campaigns) {
            if (campaigns[i]._id === campaignId) {
                  return campaigns[i].name;
            }
      }
      return null;
}

```

Reply:
Fun learning point here @adobito 
For loops have two naming conventions that I get generally tripped up by,

for (const element **of** iterable)

where you get the value

and for (iterator i **in** iterable)

where you get the iterator, int he case of arrays, the index

Reply:
Additional suggestion, return value you should be ?string, since you can return null